sdk/storage/Azure.Storage.Common/src/Shared/PartitionedUploader.cs

```diff
@@ -1,29 +1,32 @@
-﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
 using System;
 using System.Buffers;
 using System.Collections.Generic;
-using System.Diagnostics;
 using System.IO;
-using System.Runtime.CompilerServices;
 using System.Threading;
 using System.Threading.Tasks;
 using Azure.Core.Pipeline;
-using Azure.Storage.Blobs.Models;
-using Azure.Storage.Blobs.Specialized;
 using Azure.Storage.Shared;
-using Metadata = System.Collections.Generic.IDictionary<string, string>;
-using Tags = System.Collections.Generic.IDictionary<string, string>;
 
-namespace Azure.Storage.Blobs
+namespace Azure.Storage
 {
-    internal class PartitionedUploader
+    internal class PartitionedUploader<ServiceSpecificArgs, CompleteUploadReturn>
     {
+        public interface IClient
```

Comment:
This interface looks quite odd as an abstraction. Should Uploader just accept Func<> callbacks in constructor?
Alternatively could we break it down into IFullUploadingClient and IPartitionUploadingClient so that interface is more cohesive?